js/views/modals/listingDetail/Listing.js

```diff
@@ -158,8 +158,37 @@ export default class extends BaseModal {
     const photoHash = photoCol[photoIndex].original;
     const phSrc = app.getServerUrl(`ipfs/${photoHash}`);
 
-    this.$photoSection.cropit('imageSrc', phSrc);
-    this.$photoSelected.attr('data-index', photoIndex);
+    this.$photoSelected.trigger('zoom.destroy'); // old zoom must be removed
+
+    this.$photoSelectedInner
+        .one('load', this.activateZoom(photoIndex, phSrc))
+        .attr('src', phSrc);
+  }
+
+  activateZoom(photoIndex, phSrc) {
+    setTimeout(() => {
+      // without a timeout the image hasn't rendered yet and has no height or width
+
+      if (this.$photoSelectedInner.width() >= this.$photoSelectedWidth ||
+          this.$photoSelectedInner.height() >= this.$photoSelectedHeight) {
+
+        this.$photoSelected
+            .attr('data-index', photoIndex)
+            .removeClass('unzoomable')
+            .zoom({
+              url: phSrc,
+              on: 'click',
+              onZoomIn: () => {
+                this.$photoSelected.addClass('open');
+              },
+              onZoomOut: () => {
+                this.$photoSelected.removeClass('open');
+              },
+            });
+      } else {
+        this.$photoSelected.addClass('unzoomable');
+      }
+    }, 500);
```

Comment:
The timeout here is a hack, but I haven't been able to find another way around it. Even though this waits for the image load event to complete, the image hasn't rendered yet, and has a height and width of zero. 

(a timeout of 0 doesn't work, and requestAnimationFrame didn't either)

If there's another way to detect when the image has rendered, and test whether it should be zoomable at that point, I'm all ears. Possibly it would need a loop to check if the image has size yet, which seems like overkill.